File: php/features.php
<?php
declare(strict_types=1);

// UToLinkShortener SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UToLinkShortenerFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UToLinkShortenerBaseFeature();
            case "test":
                return new UToLinkShortenerTestFeature();
            default:
                return new UToLinkShortenerBaseFeature();
        }
    }
}

File: py/utolinkshortener_sdk/features.py
# UToLinkShortener SDK feature factory

from utolinkshortener_sdk.feature.base_feature import UToLinkShortenerBaseFeature
from utolinkshortener_sdk.feature.test_feature import UToLinkShortenerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UToLinkShortenerBaseFeature(),
        "test": lambda: UToLinkShortenerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
